packages/model/addon/-private/system/many-array.js

```diff
@@ -166,7 +165,25 @@ export default EmberObject.extend(MutableArray, DeprecatedEvented, {
     return false;
   },
 
+  get length() {
+    // By using `get()`, the tracking system knows to pay attention to changes that occur.
+    get(this, '[]');
+
+    if (typeof this._length === 'number') {
```

Comment:
Why do we need this check? Should we set the length in init rather than have to do this every time you access length?

Reply:
Either works!  I did that but then thought I would keep all the logic in one place.  I can do it in init if you would like!